packages/shared/lib/services/proxy.service.ts

```diff
@@ -259,7 +259,8 @@ class ProxyService {
             (error.response?.status === 403 && error.response.headers['x-ratelimit-remaining'] && error.response.headers['x-ratelimit-remaining'] === '0') ||
             error.response?.status === 429 ||
             ['ECONNRESET', 'ETIMEDOUT', 'ECONNABORTED'].includes(error.code as string) ||
-            config.retryOn?.includes(Number(error.response?.status))
+            config.retryOn?.includes(Number(error.response?.status)) ||
+            this.isRetryInHeader(config.provider.proxy?.retry, error)
```

Comment:
Just one last question (sorry my suggestion also had some shortcomings):
Are the headers always set or just when we reach rate limit? Here if its always set then all requests will be retried even on regular 400

I guess one missing check that does not exist yet in checking a "remaining headers"

Reply:
is isRetryInHeader supporting the special case for github 3 lines above?

Reply:
> is isRetryInHeader supporting the special case for github 3 lines above?

No, but I guess we could encapsulate both into updated logic. Few proposals:

1. just keep them both hard coded
```
# github
(error.response?.status === 403 && error.response.headers['x-ratelimit-remaining'] && error.response.headers['x-ratelimit-remaining'] === '0') ||

# linear 
(error.response?.status === 400 && error.response.headers['x-ratelimit-requests-remaining'] && error.response.headers['x-ratelimit-requests-remaining'] === '0') ||
```
2. Codify it in the yaml
```
# github
proxy:
        retry:
            at: 'x-ratelimit-reset'
            remaining: x-ratelimit-remaining
            error_code: 403

# linear
proxy:
           retry:
              at: 'X-RateLimit-Requests-Reset'
              remaining: x-ratelimit-requests-remaining
              error_code: 400
```

any thoughts @NangoHQ/engineers?

Reply:
the second one looks cleaner but it make providers.yaml more complex. I don't feel strongly either way

Reply:
Option 2 seems correct and not too hard to implement, I would love it 👍🏻

Reply:
Implemented option 2 with https://github.com/NangoHQ/nango/pull/3108/commits/b94e844d6144e9330c71373c28f4ffdf4feb7a08